Lib/smtplib.py

```diff
@@ -1038,7 +1038,7 @@ def _get_socket(self, host, port, timeout):
             new_socket = socket.create_connection((host, port), timeout,
                     self.source_address)
             new_socket = self.context.wrap_socket(new_socket,
-                                                  server_hostname=self._host)
+                                                  server_hostname=host)
```

Comment:
I think that the correct way is adding on `__init__()` the `self._host = host`

Reply:
I don't think so. Cause you make a instance first, the host get "" all the time. I think the best way  it's put the "self._host = host" in the connect() function. I'll try for it.

Reply:
I feel like this is probably a better solution than setting `_host` for the instance. Most of the connect and `_get_socket` methods take the host and port as input and use that instead of the one defined in the class and from that point-of-view, the use of `self._host` seems like a bug here.